Formula/lua@5.1.rb

```diff
@@ -40,14 +40,17 @@ class LuaAT51 < Formula
     end
   end
 
+  # Don't use the https://luarocks.org/releases/luarocks-x.y.z.tar.gz URL
+  # directly as it redirects to the HTTP version of the below URL.
   resource "luarocks" do
-    url "https://keplerproject.github.io/luarocks/releases/luarocks-2.3.0.tar.gz"
-    sha256 "68e38feeb66052e29ad1935a71b875194ed8b9c67c2223af5f4d4e3e2464ed97"
+    url "https://luarocks.github.io/luarocks/releases/luarocks-2.4.4.tar.gz"
+    sha256 "3938df33de33752ff2c526e604410af3dceb4b7ff06a770bc4a240de80a1f934"
   end
 
   def install
     # Use our CC/CFLAGS to compile.
     inreplace "src/Makefile" do |s|
+      s.gsub! HOMEBREW_PREFIX, prefix
```

Comment:
Does this actually work for a non-standard prefix?

Reply:
Should do, theoretically. It is just a slight variation of what we've been doing in the `lua` formula for ages, without anyone complaining about it.

Reply:
How does the replacement not fail if the prefix isn't in the file?

Reply:
Why wouldn't the prefix be in the file? We're the ones adding that line in the inline patch. Maybe I'm misunderstanding you? I've slept for about 4 hours over the last day 🙃.

Reply:
Ah, OK. That's what I get for looking at this on my iPhone.

Reply:
```
+	$(CC) -dynamiclib -install_name HOMEBREW_PREFIX/lib/liblua.5.1.dylib \
+		-compatibility_version 5.1 -current_version 5.1.5 \
+		-o liblua.5.1.5.dylib $^
```

compared to:
```
+	$(CC) -dynamiclib -install_name @LUA_PREFIX@/lib/liblua.5.3.dylib \
+		-compatibility_version 5.3 -current_version 5.3.4 \
+		-o liblua.5.3.4.dylib $^
```
in the `lua` formula. Same principle, just a slight divergence at some point for some reason. Possibly my fault given how much I've touched these formulae over the years.

Reply:
This does not look right. If I am not mistaking, this should be
```ruby
s.gsub! "HOMEBREW_PREFIX", prefix
```

Reply:
Don't think so? `brew unpack --patch lua@5.1` shows the blob as I expected it to be:

```
$(LUA_A): $(CORE_O) $(LIB_O)
	$(CC) -dynamiclib -install_name /usr/local/lib/liblua.5.1.dylib \
		-compatibility_version 5.1 -current_version 5.1.5 \
		-o liblua.5.1.5.dylib $^
```

Reply:
I get this for non-standard homebrew prefix.
```
brew upgrade lua@5.1
==> Upgrading 1 outdated package, with result:
lua@5.1 5.1.5_6
==> Upgrading lua@5.1 
==> Downloading https://www.lua.org/ftp/lua-5.1.5.tar.gz
######################################################################## 100.0%
Error: inreplace failed
src/Makefile:
  expected replacement of #<Pathname:/home/comp/chengxu/usr> with #<Pathname:/home/comp/chengxu/usr/Cellar/lua@5.1/5.1.5_6>

Reply:
I'm pretty sure I tried a string replacement in the default prefix and hit issues with it, which is why I didn't go down that route. Maybe we should just save ourselves some hassle & change it to `@LUA_PREFIX@` to match the `lua` formula.